core/src/main/java/org/springframework/security/authorization/AuthorizationObservationContext.java

```diff
@@ -69,18 +69,18 @@ public T getObject() {
 	}
 
 	/**
-	 * Get the observed {@link AuthorizationDecision}
-	 * @return the observed {@link AuthorizationDecision}
+	 * Get the observed {@link AuthorizationResult}
+	 * @return the observed {@link AuthorizationResult}
 	 */
-	public AuthorizationDecision getDecision() {
+	public AuthorizationResult getDecision() {
```

Comment:
To remain [binary compatible](https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_API_interfaces_-_API_methods), we cannot change return types. Instead, please add a new getter (called `getAuthorizationResult`) and deprecate this getter.